### PR TITLE
add resource detectors to spring starter

### PR DESCRIPTION
**Pull request by zeitlinger**
it's a lightweight dependency and very useful to get started

### CODE REVIEW

File: instrumentation/spring/starters/spring-boot-starter/build.gradle.kts
Comment (laurit) [inline instrumentation/spring/starters/spring-boot-starter/build.gradle.kts:15]: why did you choose to add this as `implementation` when all the other dependencies are `api`?
secondly these resource providers use the SPI that the sdk autoconfigure module uses, but spring boot starter does not uses the sdk autoconfigure. Even if you add them do they actually do anything?

Comment (jeanbisutti) [reply instrumentation/spring/starters/spring-boot-starter/build.gradle.kts:15]: I would see more the new dependency included in `spring:spring-boot-autoconfigure`.

Comment (zeitlinger) [reply instrumentation/spring/starters/spring-boot-starter/build.gradle.kts:15]: 1. implementation is more conservative (doesn't allow access to transitive dependencies), so I use that by default
2. .. does not use autoconfigure ... -> I've been wondering the same - is there there a good reason for that?
3. will it work? Yes, there's a [guard that checks if the class exists](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/OtelResourceAutoConfiguration.java#L48)

Comment (jeanbisutti) [reply instrumentation/spring/starters/spring-boot-starter/build.gradle.kts:15]: @zeitlinger Could you develop one feature for OTel Spring autoconfiguration with this dependency and with one test?

Comment (zeitlinger) [reply instrumentation/spring/starters/spring-boot-starter/build.gradle.kts:15]: > I would see more the new dependency included in `spring:spring-boot-autoconfigure`.

autoconfigure modules should be as thin as possible, and starters add what's usually needed.

Comment (jeanbisutti) [reply instrumentation/spring/starters/spring-boot-starter/build.gradle.kts:15]: The resource library is [compileOnly in Spring autoconfiguration](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/9e208c85740dfa5035c29ba84a419ff94b9ee519/instrumentation/spring/spring-boot-autoconfigure/build.gradle.kts#L49). Do we want to include it by default?

**Review by jeanbisutti — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Comment by jeanbisutti** [comment]
@laurit We could merge this one as discussed during the last SIG meeting? I could follow up with a PR proposing to move some of  the OTel starter dependencies to the autoconfiguration library.

**Comment by jeanbisutti** [comment]
> @jeanbisutti I didn't participate in the last SIG meeting, feel free to proceed according to what you agreed at the SIG meeting. If we are going to include the resource detectors I think we should also consider including https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/spring/spring-boot-resources/library

@laurit It seems I can't merge

**Comment by laurit** [comment]
@jeanbisutti you should be able to merge this, if you can't merge other approved PRs then we need to figure our why it is so

**Comment by trask** [comment]
@laurit this is intentional (though we didn't have this repo set up correctly way back when you were an approver):

https://github.com/open-telemetry/community/blob/main/docs/how-to-configure-new-repository.md#branch-protection-rule-main

> * Restrict who can push to matching branches: ✔️
>   * Organization administrators, repository administrators, and users with the Maintain role

**Comment by jeanbisutti** [comment]
> @jeanbisutti you should be able to merge this, if you can't merge other approved PRs then we need to figure our why it is so

@laurit Same thing for #10350, for which you have approved and I have not

**Comment by zeitlinger** [comment]
> @laurit It seems I can't merge

I also can't merge approved PRs

**Pull request merged**
